src/Middleware/ExceptionResponder.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Middleware;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Injector\Injector;
+
+class ExceptionResponder implements MiddlewareInterface
+{
+    private array $exceptionMap;
+    private ResponseFactoryInterface $responseFactory;
+    private Injector $injector;
+
+    /**
+     * @param array $exceptionMap
+     * @psalm-param array{string, int|callable}
+     */
+    public function __construct(array $exceptionMap, ResponseFactoryInterface $responseFactory, Injector $injector)
```

Comment:
You can specify the ContainerInterface in the parameters and get the ResponseFactoryInterface and Injector already in the catch body

Reply:
I don't see a profit in doing it. Makes code less clean.

Reply:
Mb. I just don't like that you pass the exception map along with the dependencies.